main/src/ca/uwaterloo/flix/language/ast/TypeConstructor.scala

```diff
@@ -86,14 +86,48 @@ object TypeConstructor {
     def kind: Kind = Kind.Star
   }
 
+  /**
+    * A type constructor that represents the type of empty records.
+    */
+  case object RecordEmpty extends TypeConstructor {
+    def kind: Kind = Kind.Record
+  }
+
+  /**
+    * A type constructor that represents the type of extended records.
+    */
+  case class RecordExtend(label: String) extends TypeConstructor {
+    /**
+      * The shape of an extended record is Record[t;r]
+      */
```

Comment:
I would remove this comment or write something that more closely matches the surface syntax of records, e.g. `{ label = value | rest }`